src/boot/efi/random-seed.c

```diff
@@ -14,15 +14,28 @@
 
 #define EFI_RNG_GUID &(const EFI_GUID) EFI_RNG_PROTOCOL_GUID
 
+struct linux_efi_random_seed {
+      uint32_t size;
+      unsigned char seed[];
+};
+
+#define LINUX_EFI_RANDOM_SEED_TABLE_GUID \
+        &(EFI_GUID) { 0x1ce1e5bc, 0x7ceb, 0x42f2,  { 0x81, 0xe5, 0x8a, 0xad, 0xf1, 0x80, 0xf5, 0x7b } }
+
 /* SHA256 gives us 256/8=32 bytes */
 #define HASH_VALUE_SIZE 32
 
-static EFI_STATUS acquire_rng(UINTN size, void **ret) {
-        _cleanup_free_ void *data = NULL;
+/* Linux's RNG is 256 bits, so let's provide this much */
+#define DESIRED_SEED_SIZE 32
+
+/* Some basic domain separation in case somebody uses this data elsewhere */
+#define HASH_LABEL "systemd-boot random seed label v1"
+
+static EFI_STATUS acquire_rng(void *out, UINTN size) {
```

Comment:
so we have this rule in systemd that return parameters should always be called "ret" or "ret_xyz". I think we should follow that here too. i.e. "out" → "ret"

Reply:
Will do.